docs/advanced-features/athena.md

```diff
@@ -1,91 +1,475 @@
+
+### A. Create a new Athena policy.
+
+The standard [AmazonAthenaFullAccess](https://console.aws.amazon.com/iam/home#/policies/arn:aws:iam::aws:policy/AmazonAthenaFullAccess) policy is more permissive than necessary.  For production usage, we recommend creating a policy limited to only the above Bucket:
+<!--pytest-codeblocks:cont-->
+
+
+```python
+# https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/iam.html#IAM.ServiceResource.create_policy
+# https://docs.aws.amazon.com/athena/latest/ug/workgroups-access.html
+
+AthenaQuiltAccess = {
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "GrantQuiltAthenaFullAccess",
+            "Effect": "Allow",
+            "Action": [
+                "athena:ListWorkGroups",
+            ],
+            "Resource": [
+                "*",
+            ],
+        },
+        {
+            "Sid": "GrantQuiltAthenaAccess",
+            "Effect": "Allow",
+            "Action": [
+                "athena:Create*",
+                "athena:Get*",
+                "athena:ListDatabases",
+                "athena:ListNamedQueries",
+                "athena:ListPreparedStatements",
+                "athena:ListQueryExecutions",
+                "athena:StartQueryExecution",
+                #                "athena:Update*",
+            ],
+            "Resource": [
+                ARN_CATALOG,
+                ARN_DATACATALOG,
+                ARN_WORKGROUP,
+            ],
+        },
+        {
+            "Sid": "GrantQuiltGlueAccess",
+            "Effect": "Allow",
+            "Action": [
+                "glue:GetDatabase",
+                "glue:GetDatabases",
+                "glue:GetTable",
+                "glue:GetTables",
+            ],
+            "Resource": [
+                "*",
+            ],
+        },
+        {
+            "Sid": "GrantQuiltGlueWriteAccess",
+            "Effect": "Allow",
+            "Action": [
+                "glue:CreateTable",
+                "glue:DeleteTable",
+                "glue:UpdateTable",
+            ],
+            "Resource": [
+                ARN_CATALOG,
+                ARN_DATABASE,
+                ARN_TABLE + "/*_quilt_*",
+            ],
+        },
+        {
+            "Sid": "GrantQuiltAthenaBucketAccess",
+            "Effect": "Allow",
+            "Action": [
+                "s3:GetBucketLocation",
+                "s3:ListBucket",
+                "s3:ListBucketMultipartUploads",
+                "s3:ListMultipartUploadParts",
+            ],
+            "Resource": [
+                "*",
+            ],
+        },
+        {
+            "Sid": "GrantQuiltAthenaInputAccess",
+            "Effect": "Allow",
+            "Action": [
+                "s3:GetObject",
+            ],
+            "Resource": [
+                "arn:aws:s3:::*",
+                "arn:aws:s3:::*/.quilt/*",
```

Comment:
Isn't the second one is a subset of the first one?

Reply:
I don't think so. The first grants a view OF the bucket, the second is INSIDE a specific folder. Its a weird convention, to be sure.

Reply:
> The first grants a view OF the bucket, the second is INSIDE a specific folder.

I'm not really sure what you mean here, but I tested it and still think that first is a superset of the second.